typescript/infra/scripts/check/check-validator-version.ts

```diff
@@ -0,0 +1,206 @@
+import { ValidatorAnnounce__factory } from '@hyperlane-xyz/core';
+import {
+  ChainName,
+  S3Validator,
+  defaultMultisigConfigs,
+} from '@hyperlane-xyz/sdk';
+
+import { isEthereumProtocolChain } from '../../src/utils/utils.js';
+import { getArgs, withChains } from '../agent-utils.js';
+import { getEnvironmentConfig, getHyperlaneCore } from '../core-utils.js';
+
+const acceptableValidatorVersions: Record<string, string> = {
+  '72d498fa984750b9137c1211fef6c80a3e594ce7': 'aug-27-batch', // Aug 27 deploy
+  d71dd4e5ed7eb69cc4041813ef444e37d881cdda: 'sep-9-batch', // Sep 9 deploy
+  '45399a314cec85723bbb5d2360531c96a3aa261e': 'oct-27-batch', // Oct 27 deploy
+  '75d62ae7bbdeb77730c6d343c4fc1df97a08abe4': 'nov-7-batch', // Nov 7 deploy
+  a64af8be9a76120d0cfc727bb70660fa07e70cce: 'pre-1.0.0', // pre-1.0.0
+  ffbe1dd82e2452dbc111b6fb469a34fb870da8f1: '1.0.0', // 1.0.0
+  '79453fcd972a1e62ba8ee604f0a4999c7b938582': 'tesselated-special-build', // Tessellated's Own Build
+};
+
+// TODO: refactor multisigIsm.ts to include mappings of addresses to aliases as part of the config
+// This will also allow us to programmatically generate the default ISM docs page.
+const KNOWN_VALIDATOR_ADDRESSES: Record<string, string> = {
```

Comment:
I vote we change the multisigIsm.ts format in the sdk to be something like
```
{ validator: string, name: string }
```
It's a commonly requested feature and is bound to be useful elsewhere, also less likely to go stale

Reply:
made this a separate PR https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4896